### PR TITLE
Remove gh-pages deploy in favor of netlify

### PR DESCRIPTION
**Pull request by shilman**
Issue: #4126 

## What I did

- Remove gh-pages deploy from circle config
- Keep the docs build just so we can be sure that it builds
- Keep the deploy step as a placeholder

Now the gh-pages branch will go out of date (and we should probably remove it once we are fully onto netlify), and storybook.js.org points to the netlify instance rather than to storybooks.github.io/storybook (See https://github.com/js-org/js.org/pull/2304)

## How to test

Not really testable until we merge this into a tag `vX.Y.Z` commit. But I've been messing around with this for an hour or so, and am confident it's safe.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -287,10 +287,7 @@ jobs:
       - run:
           name: Deploy
           command: |
-            cd docs
-            git config --global user.email "gh-pages@localhost"
-            git config --global user.name "npm gh-pages"
-            yarn deploy:ci
+            echo "Nothing to deploy"
```
Comment (pksunkara) [inline .circleci/config.yml]: You can remove the whole job.

**Review by ndelangen — APPROVED** [review]
please remove the whole job from config

**Review by pksunkara — CHANGES_REQUESTED** [review]
https://github.com/storybooks/storybook/issues/4126#issuecomment-419094402

**Comment by pksunkara** [comment]
@shilman Before merging this PR, could you please address this? https://github.com/storybooks/storybook/issues/4126#issuecomment-419094402. We need to figure out what process we are going to follow.

**Review by pksunkara — APPROVED** [review]

**Comment by pksunkara** [comment]
I actually realized that the current deploy job is broken anyway because of circleci tag filter rule. So, I am merging this.

**Pull request merged**